jta-1_2-standalone/src/main/java/org/jboss/narayana/quickstarts/jta/RequiredCounterManager.java

```diff
@@ -16,37 +16,105 @@
  */
 package org.jboss.narayana.quickstarts.jta;
 
+import javax.enterprise.context.Destroyed;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.Observes;
+import javax.enterprise.event.TransactionPhase;
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.spi.BeanManager;
 import javax.inject.Inject;
-import javax.naming.InitialContext;
-import javax.naming.NamingException;
+import javax.transaction.Status;
+import javax.transaction.SystemException;
+import javax.transaction.TransactionManager;
+import javax.transaction.TransactionScoped;
 import javax.transaction.Transactional;
 import javax.transaction.UserTransaction;
 
+import org.jboss.logging.Logger;
+import org.jboss.weld.transaction.spi.TransactionServices;
+
 /**
+ * <p>
+ * A class with definition of the
+ * {@link Transactional.TxType#REQUIRED} transactional boundary
+ * for one particular method {@link #isTransactionAvailable()}.
+ * </p>
+ * <p>
+ * If the method is invoked with a transactional context being
+ * available the method joins the context.
+ * If there is no context available a new transactional context
+ * is created (a new transaction is started) before the method
+ * code is executed.
+ * </p>
+ * <p>
+ * The class demonstrates the usage of the {@link TransactionScoped} events.
+ * A method can observe the {@link Initialized} and {@link Destroyed} events.
+ * </p>
+ * 
  * @author <a href="mailto:gytis@redhat.com">Gytis Trikleris</a>
  */
-@Transactional
 public class RequiredCounterManager {
+    private static final Logger LOG = Logger.getLogger(RequiredCounterManager.class);
 
     @Inject
     private Counter counter;
 
+    @Inject
+    private EventsCounter lifeCycle;
+
+    @Inject
+    private TransactionManager transactionManager;
+
+    // available only when Weld TransactionServices are implemented
+    // @Inject
+    // private UserTransaction userTransaction;
```

Comment:
Maybe remove the commented out code?

Reply:
This addition was intentional for user starting with the quickstart. If they want to use the `UserTransaction` they need ensure the `TransactionServices` SPI interface is implemented. This was meant as a kind of reminder. If you think it should not be there (not being done this way) I can remove it.

Reply:
I see - thanks for info, it's OK to do this way